src/re_id_tracker.py

```diff
@@ -0,0 +1,179 @@
+        """Validate config and returns a list of dependencies."""
+        camera_name = config.attributes.fields["camera_name"].string_value
+        _ = ReIDObjetcTrackerConfig(config)
+        return [camera_name]
+
+    def reconfigure(
+        self, config: ServiceConfig, dependencies: Mapping[ResourceName, ResourceBase]
+    ):
+        camera_name = config.attributes.fields["camera_name"].string_value
+        self.camera = dependencies[Camera.get_resource_name(camera_name)]
+
+        re_id_tracker_cfg = ReIDObjetcTrackerConfig(config)
+        if self.tracker is not None:
+            create_task(self.tracker.stop())
+        self.tracker = Tracker(re_id_tracker_cfg, camera=self.camera)
+
+    async def get_properties(
+        self,
+        *,
+        extra: Optional[Mapping[str, Any]] = None,
+        timeout: Optional[float] = None,
+    ) -> Vision.Properties:
+        return Vision.Properties(
+            classifications_supported=True,
+            detections_supported=True,
+            object_point_clouds_supported=False,
+        )
+
+    async def capture_all_from_camera(
+        self,
+        camera_name: str,
+        return_image: bool = False,
+        return_classifications: bool = False,
+        return_detections: bool = False,
+        return_object_point_clouds: bool = False,
+        *,
+        extra: Optional[Mapping[str, Any]] = None,
+        timeout: Optional[float] = None,
+    ):
+        img = None
+        if return_image:
+            img = await self.camera.get_image(mime_type=CameraMimeType.JPEG)
+
+        classifications = None
+        if return_classifications:
+            if await self.tracker.is_new_object_detected():
+                classifications = [
+                    Classification(class_name="new_object_detected", confidence=1)
+                ]
+
+        detections = None
+        if return_detections:
+            detections = await self.tracker.get_current_detections()
+        return CaptureAllResult(
+            image=img, classifications=classifications, detections=detections
+        )
+
+    async def get_object_point_clouds(
+        self,
+        camera_name: str,
+        *,
+        extra: Optional[Dict[str, Any]] = None,
+        timeout: Optional[float] = None,
+        **kwargs,
+    ) -> List[PointCloudObject]:
+        raise NotImplementedError
+
+    async def get_detections(
+        self,
+        image: ViamImage,
+        *,
+        extra: Mapping[str, Any],
+        timeout: float,
+    ) -> List[Detection]:
+        return await self.tracker.get_current_detections()
+
+    async def get_classifications(
+        self,
+        image: ViamImage,
+        count: int,
+        *,
+        extra: Optional[Mapping[str, Any]] = None,
+        timeout: Optional[float] = None,
+    ) -> List[Classification]:
+        if await self.tracker.is_new_object_detected():
+            return [Classification(class_name="new_object_detected", confidence=1)]
```

Comment:
It's a TODO to decide what should be the confidence. I think it should be something like the inverse of the distance more than the confidence of the detector.